src/internal_events/aws_s3.rs

```diff
@@ -81,33 +84,93 @@ pub mod source {
     }
 
     #[derive(Debug)]
-    pub(crate) struct SqsMessageDeleteSucceeded<'a> {
-        pub message_id: &'a str,
+    pub(crate) struct SqsMessageDeleteSucceeded {
+        pub message_ids: Vec<DeleteMessageBatchResultEntry>,
     }
 
-    impl<'a> InternalEvent for SqsMessageDeleteSucceeded<'a> {
+    impl InternalEvent for SqsMessageDeleteSucceeded {
         fn emit_logs(&self) {
-            trace!(message = "Deleted SQS message.", %self.message_id);
+            let message_ids = self
+                .message_ids
+                .iter()
+                .map(|x| x.id.to_string())
+                .collect::<Vec<_>>()
+                .join(", ");
+            trace!(message = "Deleted SQS message(s).", %message_ids);
         }
 
         fn emit_metrics(&self) {
-            counter!("sqs_message_delete_succeeded_total", 1);
+            counter!(
+                "sqs_message_delete_succeeded_total",
+                self.message_ids.len() as u64
+            );
         }
     }
 
     #[derive(Debug)]
-    pub(crate) struct SqsMessageDeleteFailed<'a> {
-        pub message_id: &'a str,
-        pub error: &'a RusotoError<DeleteMessageError>,
+    enum MessageDeleteFailureState {
+        Complete(
+            Vec<DeleteMessageBatchRequestEntry>,
+            RusotoError<DeleteMessageBatchError>,
+        ),
+        Partial(Vec<BatchResultErrorEntry>),
+    }
+
+    #[derive(Debug)]
+    pub(crate) struct SqsMessageDeleteFailed {
+        state: MessageDeleteFailureState,
```

Comment:
The code paths for `Complete` and `Partial` below seem to be completely disjoint. What is the benefit of making this an enum instead of two separate event structs?

Reply:
Continuity in downstream systems that currently look at these events was the thought... although maybe that doesn't really matter?

Reply:
I'm not sure how. There are already two separate code paths that emit the same warning and metric (albeit in the same function). I don't think separating them into separate methods/structs will make that big a deal (assuming I understand your concern correctly).